ignite/contrib/metrics/average_precision.py

```diff
@@ -22,6 +22,8 @@ class AveragePrecision(EpochMetric):
             :class:`~ignite.engine.Engine`'s `process_function`'s output into the
             form expected by the metric. This can be useful if, for example, you have a multi-output model and
             you want to compute the metric with respect to one of the outputs.
+        check_compute_fn (bool): if True, compute_fn is run on the first batch of data to ensure there are no
+            issues. If issues exist, user is warned that there might be an issue with the compute_fn.
```

Comment:
Maybe, it wont be clear about which `compute_fn` it says.

Reply:
Could you clarify Victor?

Reply:
Right, `AveragePrecision` has no `compute_fn` as argument and it executes `average_precision_score` internally. This argument `check_compute_fn` as described right now can be missleading.

I was also thinking that for those metrics like AP, ROC etc it would be more likely to see warnings as a single batch is not that representative... Another thought was to make it False by default or even always False...

Reply:
When implementing it - that's what I was thinking - although a sanity check is a good idea IMO. Keeping it True will keep the behavior consistent across versions.

Reply:
Do you mean keeping it False for just those metrics (AP, ROC, etc)?

Reply:
True, but somehow it causes this issue : #1103 and slowdown of computations (should be confirmed though) ... 
OK, let's keep it True and just improve docstring

Reply:
> Do you mean keeping it False for just those metrics (AP, ROC, etc)?

Yes, 
- False for only for AP, ROC etc 
- True for `EpochMetric`

Reply:
What are your thoughts on keeping it False and not giving the user an option to change it?

Reply:
> What are your thoughts on keeping it False and not giving the user an option to change it for these contrib metrics?

Let's keep it False and with option to change to True.

Reply:
Thoughts on the docstring below?

check_compute_fn (bool): Optional default False. If True, ``sklearn.metrics.average_precision_score`` is run on the first batch of data to ensure there are no issues. User will be warned in case there are any issues computing the function.

Reply:
Looks good ! Please, add `sklearn.metrics.average_precision_score` as link to sklearn docs.

Reply:
Done!